### DISCUSSION TITLE
Interleaved buffer + TextLayer

### DISCUSSION THREAD

**Discussion by brandonjpierce** [visgl/deck.gl#6633] [Q&A] [answered]
Hello, 

I am attempting to use an interleaved buffer per the [example here](https://github.com/visgl/deck.gl/tree/master/examples/experimental/interleaved-buffer) however I am running into an issue when using this buffer for a TextLayer. 

Here is a codesandbox example: https://codesandbox.io/s/currying-butterfly-1r3qg

As you can see the ScatterplotLayer renders great but the TextLayer is throwing an error: 

```
codes.subarray is not a function
```

Any ideas or recommendations?

**Comment by Pessimistress** [comment, marked as answer]
https://deck.gl/docs/api-reference/layers/text-layer#use-binary-attributes

**Reply by brandonjpierce** [reply]
This example assumes separate buffers per attribute, in the codesandbox example I linked I am using a single interleaved buffer

**Reply by Pessimistress** [reply]
Your interleaved buffer still needs to have one set of attributes per character. Your buffer has one position/color for each label.

**Reply by brandonjpierce** [reply]
Ohhhhh right:

> The position attribute must supply one position for each character

That is the culprit then for sure. Thanks!
